ironfish/src/rpc/routes/wallet/createTransaction.ts

```diff
@@ -122,10 +124,16 @@ routes.register<typeof CreateTransactionRequestSchema, CreateTransactionResponse
       params.outputs = []
 
       for (const output of request.data.outputs) {
+        const memo = output.memo
+          ? Buffer.from(output.memo)
+          : output.memoHex
+          ? Buffer.from(output.memoHex, 'hex')
+          : Buffer.alloc(32, 0)
```

Comment:
We should throw if both are null right? This Buffer.alloc would lead to a cryptic error in the case of both being null.

Reply:
I think the empty Buffer should be fine here -- the resulting memo would be the same as if the user submitted an empty string for the memo.

I'll add a test to cover this case

Reply:
Related: what if both are non-null/non-empty? Shouldn't we throw in that case?

Reply:
> So it is acceptable to send no memo at all? Because previously memo was a required RPC field (albeit could be sent with empty string).

I think it makes for a nicer interface not to have to provide an empty string, but I'm open to reverting it back to a required field

Reply:
> Related: what if both are non-null/non-empty? Shouldn't we throw in that case?

I think we should throw in that case -- the client shouldn't set both

Reply:
The change here is to make the RPC not required to take a memo. It's not a necessary input to creating a transaction and was really just rude to require users to put as an input.

Reply:
Ok that seems reasonable to me, this is just a change in behavior that isn’t listed in PR summary